test/test262/errors.js

```diff
@@ -11,123 +12,109 @@ function parse(src, isModule, earlyErrors) {
 
 suite('test262', () => {
   suite('pass and pass-explicit', () => {
-    let passExcludes = [
-      // https://github.com/shapesecurity/shift-parser-js/issues/311
-      '995.script.js',
-
-      // This is an invalid test
-      '970.script.js',
-
-      // https://github.com/shapesecurity/shift-parser-js/issues/311
-      '1012.script.js',
-
-      // https://github.com/tc39/test262-parser-tests/issues/12
-      '1252.script.js',
-    ];
-    fs.readdirSync(`${scriptDir}/pass`)
-      .filter(item => passExcludes.indexOf(item) === -1)
-      .forEach(f => {
-        let passTree, passLocations, passExplicitTree;
-        let passTestDir = `${scriptDir}/pass/${f}`;
-        let passExplicitTestDir = `${scriptDir}/pass-explicit/${f}`;
-        test(`does not throw error and generates same tree[${f}]`, () => {
-          let passSrc = fs.readFileSync(passTestDir, 'utf8');
-          expect(() => {
-            ({ tree: passTree, locations: passLocations } = parse(passSrc, f.match('.module.js'), true));
-          }).to.not.throwError();
+    for (let f of fs.readdirSync(`${scriptDir}/pass`)) {
+      if (expectations.xfail.pass.indexOf(f) !== -1) {
+        continue;
+      }
+
+      let passTestFile = `${scriptDir}/pass/${f}`;
+      let passExplicitTestFile = `${scriptDir}/pass-explicit/${f}`;
+      test(`does not throw error and generates same tree[${f}]`, () => {
+        let passSrc = fs.readFileSync(passTestFile, 'utf8');
+        let passTree, passLocations;
+        expect(() => {
+          ({ tree: passTree, locations: passLocations } = parse(passSrc, f.match('.module.js'), true));
+        }).to.not.throwError();
+        locationSanityCheck(passTree, passLocations);
+
+        let passExplicitSrc = fs.readFileSync(passExplicitTestFile, 'utf8');
+        let passExplicitTree;
+        expect(() => {
+          passExplicitTree = parse(passExplicitSrc, f.match('.module.js'), true).tree;
```

Comment:
Can you extract all of these `f.match` calls to an `isModule` function?

Reply:
Done.